libraries/chain/controller.cpp

```diff
@@ -3296,7 +3296,7 @@ struct controller_impl {
 
             if (!my_finalizers.empty()) {
                block_handle_accessor::apply_s<void>(chain_head, [&](const auto& head) {
-                  if (head->is_recent() || my_finalizers.is_active()) {
+                  if (head->is_recent() || testing_allow_voting) {
```

Comment:
This doesn't seem right. By default `testing_allow_voting` is true, which means that by default we allow voting on old blocks.